util/encoding/encoding.go

```diff
@@ -398,12 +399,20 @@ func EncodeBytes(b []byte, data []byte) []byte {
 	return append(b, escape1, escapedTerm)
 }
 
+// EncodeBytesDecreasing encodes the []byte value using an
```

Comment:
The escape-based encoding isn't why it sorts in reverse order. Maybe reword as:

"encodes the []byte value using an escape-based encoding and then ones-complement so that it sorts in reverse order"